File: tests/fixtures/sample_map_dispatch.py
# Fixture for testing REFERENCES edge extraction in Python map dispatch patterns.


def handle_create(data):
    print("create", data)


def handle_update(data):
    print("update", data)


def handle_delete(data):
    print("delete", data)


def validate_input(data):
    return data is not None


def process_data(data):
    return data


def format_output(data):
    return str(data)


# Pattern 1: Dict with function values
handlers = {
    "create": handle_create,
    "update": handle_update,
    "delete": handle_delete,
}

# Pattern 2: List of function references (pipeline)
pipeline = [validate_input, process_data, format_output]


# Pattern 3: Assignment to dict key
dynamic_handlers = {}
dynamic_handlers["format"] = format_output


def dispatch(action):
    handler = handlers.get(action)
    if handler:
        handler({})

File: tests/fixtures/sample_map_dispatch.ts
// Fixture for testing REFERENCES edge extraction in map dispatch patterns.

function handleCreate(data: any): void {
    console.log("create", data);
}

function handleUpdate(data: any): void {
    console.log("update", data);
}

function handleDelete(data: any): void {
    console.log("delete", data);
}

function validateInput(data: any): boolean {
    return data != null;
}

function processData(data: any): any {
    return data;
}

function formatOutput(data: any): string {
    return JSON.stringify(data);
}

// Pattern 1: Object literal with function values (Record<string, Handler>)
const handlers: Record<string, (data: any) => void> = {
    create: handleCreate,
    update: handleUpdate,
    delete: handleDelete,
};

// Pattern 2: Shorthand property references
const shorthandMap = { validateInput, processData };

// Pattern 3: Property assignment to map
const dynamicHandlers: Record<string, Function> = {};
dynamicHandlers['format'] = formatOutput;

// Pattern 4: Array of function references (pipeline)
const pipeline = [validateInput, processData, formatOutput];

// Pattern 5: Function passed as callback argument
function register(fn: Function): void {
    // registration logic
}

function dispatch(action: string): void {
    const handler = handlers[action];
    if (handler) {
        register(handleCreate);
        handler({});
    }
}
